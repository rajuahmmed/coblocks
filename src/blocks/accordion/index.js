/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	attributes,
	transforms,
	edit,
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
};

export { name, category, metadata, settings };
