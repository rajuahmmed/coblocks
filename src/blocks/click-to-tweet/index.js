/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import deprecated from './deprecated';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Click to Tweet', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a quote for readers to tweet via Twitter.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'share', 'coblocks' ),
		/* translators: block keyword */
		__( 'twitter', 'coblocks' ),
	],
	example: {
		attributes: {
			content: __( 'The easiest way to promote and advertise your blog, website, and business on Twitter.', 'coblocks' ),
		},
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
