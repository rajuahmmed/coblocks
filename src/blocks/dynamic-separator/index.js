/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Dynamic HR', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a resizable spacer between other blocks.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		'hr',
		/* translators: block keyword */
		__( 'spacer', 'coblocks' ),
	],
	styles: [
		{
			name: 'dots',
			/* translators: block style */
			label: __( 'Dot', 'coblocks' ),
			isDefault: true,
		},
		{
			name: 'line',
			/* translators: block style */
			label: __( 'Line', 'coblocks' ),
		},
		{
			name: 'fullwidth',
			/* translators: block style */
			label: __( 'Fullwidth', 'coblocks' ),
		},
	],
	example: {
		attributes: {
			height: 100,
		},
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
