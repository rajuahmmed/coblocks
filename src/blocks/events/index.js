/**
 * Internal dependencies.
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies.
 */
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	title: _x( 'Events', 'block name', 'coblocks' ),
	description: __( 'Add a list of events or display events from a public calendar.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'calendar', 'coblocks' ),
		/* translators: block keyword */
		__( 'date', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
	},
	transforms,
	edit,
	save,
};

export { metadata, name, category, settings };
