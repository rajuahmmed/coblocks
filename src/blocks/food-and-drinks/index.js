/**
 * Internal dependencies.
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import deprecated from './deprecated';

/**
 * WordPress dependencies.
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Food & Drinks', 'coblocks' ),
	/* translators: block description */
	description: __( 'Display a menu or price list.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'restaurant', 'coblocks' ),
		/* translators: block keyword */
		__( 'menu', 'coblocks' ),
	],
	supports: {
		align: [ 'wide' ],
	},
	getEditWrapperProps( attributes ) {
		return { 'data-columns': attributes.columns };
	},
	deprecated,
	attributes,
	edit,
	save,
};

export { name, category, metadata, settings };
