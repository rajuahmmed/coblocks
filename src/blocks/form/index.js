/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import variations from './variations';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const metadata = {
	name: 'coblocks/form',
	category: 'layout',
	attributes: {
		subject: {
			type: 'string',
			default: null,
		},
		to: {
			type: 'string',
			default: null,
		},
	},
};

const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Form', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a contact form to your page.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'email', 'coblocks' ),
		/* translators: block keyword */
		__( 'about', 'coblocks' ),
		/* translators: block keyword */
		__( 'contact', 'coblocks' ),
	],
	supports: {
		reusable: false,
		html: false,
		customClassName: false,
	},
	example: {
		attributes: {
			subject: __( 'Subject example', 'coblocks' ),
		},
	},
	attributes,
	variations,
	edit,
	save: InnerBlocks.Content,
};

export { name, category, metadata, settings };
