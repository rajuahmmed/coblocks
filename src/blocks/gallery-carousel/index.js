/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import { GalleryAttributes } from '../../components/block-gallery/shared';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...GalleryAttributes,
	...metadata.attributes,
};

const settings = {
	/* translators: block name */
	title: __( 'Carousel', 'coblocks' ),
	/* translators: block description */
	description: __( 'Display multiple images in a beautiful carousel gallery.', 'coblocks' ),
	attributes,
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'gallery', 'coblocks' ),
		/* translators: block keyword */
		__( 'photos', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
		coBlocksSpacing: true,
	},
	example: {
		attributes: {
			gridSize: 'lrg',
			gutter: 5,
			images: [
				{ url: 'https://s.w.org/images/core/5.3/Biologia_Centrali-Americana_-_Cantorchilus_semibadius_1902.jpg' },
				{ url: 'https://s.w.org/images/core/5.3/Glacial_lakes,_Bhutan.jpg' },
				{ url: 'https://s.w.org/images/core/5.3/MtBlanc1.jpg' },
			],
		},
	},
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, icon, metadata, settings };
