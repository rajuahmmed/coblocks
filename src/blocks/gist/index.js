/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Gist', 'coblocks' ),
	/* translators: block description */
	description: __( 'Embed GitHub gists by adding a gist link.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		'github',
		/* translators: block keyword */
		__( 'code', 'coblocks' ),
	],
	supports: {
		html: false,
		align: [ 'wide' ],
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
