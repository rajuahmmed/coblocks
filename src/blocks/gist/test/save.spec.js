/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( 'coblocks/gist', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render with url attribute', () => {
		block.attributes.url = 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with meta attribute', () => {
		block.attributes.url = 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7';
		block.attributes.meta = false;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7' );
		expect( serializedBlock ).toContain( 'no-meta' );
		expect( serializedBlock ).toContain( '"meta":false' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with file attribute', () => {
		block.attributes.url = 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7';
		block.attributes.file = 'file.js';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7' );
		expect( serializedBlock ).toContain( 'file=file.js' );
		expect( serializedBlock ).toContain( '"file":"file.js"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with caption attribute', () => {
		block.attributes.url = 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7';
		block.attributes.caption = 'caption';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7' );
		expect( serializedBlock ).toContain( '<figcaption>caption</figcaption>' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with align attribute', () => {
		block.attributes.url = 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7';
		block.attributes.align = 'wide';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7' );
		expect( serializedBlock ).toContain( 'alignwide' );
		expect( serializedBlock ).toContain( '"align":"wide"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with className attribute', () => {
		block.attributes.url = 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7';
		block.attributes.className = 'my-custom-class';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7' );
		expect( serializedBlock ).toContain( 'my-custom-class' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
