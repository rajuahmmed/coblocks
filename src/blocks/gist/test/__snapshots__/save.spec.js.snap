// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/gist should render with align attribute 1`] = `
"<!-- wp:coblocks/gist {\\"url\\":\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\",\\"align\\":\\"wide\\"} -->
<div class=\\"wp-block-coblocks-gist alignwide\\"><script src=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7.js\\"></script><noscript><a href=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\">View this gist on GitHub</a></noscript></div>
<!-- /wp:coblocks/gist -->"
`;

exports[`coblocks/gist should render with caption attribute 1`] = `
"<!-- wp:coblocks/gist {\\"url\\":\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\"} -->
<div class=\\"wp-block-coblocks-gist\\"><script src=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7.js\\"></script><noscript><a href=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\">View this gist on GitHub</a></noscript><figcaption>caption</figcaption></div>
<!-- /wp:coblocks/gist -->"
`;

exports[`coblocks/gist should render with className attribute 1`] = `
"<!-- wp:coblocks/gist {\\"url\\":\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\",\\"className\\":\\"my-custom-class\\"} -->
<div class=\\"wp-block-coblocks-gist my-custom-class\\"><script src=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7.js\\"></script><noscript><a href=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\">View this gist on GitHub</a></noscript></div>
<!-- /wp:coblocks/gist -->"
`;

exports[`coblocks/gist should render with file attribute 1`] = `
"<!-- wp:coblocks/gist {\\"url\\":\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\",\\"file\\":\\"file.js\\"} -->
<div class=\\"wp-block-coblocks-gist\\"><script src=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7.js?file=file.js\\"></script><noscript><a href=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7#file-file-js\\">View this gist on GitHub</a></noscript></div>
<!-- /wp:coblocks/gist -->"
`;

exports[`coblocks/gist should render with meta attribute 1`] = `
"<!-- wp:coblocks/gist {\\"url\\":\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\",\\"meta\\":false} -->
<div class=\\"wp-block-coblocks-gist no-meta\\"><script src=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7.js\\"></script><noscript><a href=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\">View this gist on GitHub</a></noscript></div>
<!-- /wp:coblocks/gist -->"
`;

exports[`coblocks/gist should render with url attribute 1`] = `
"<!-- wp:coblocks/gist {\\"url\\":\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\"} -->
<div class=\\"wp-block-coblocks-gist\\"><script src=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7.js\\"></script><noscript><a href=\\"https://gist.github.com/someuser/a04f4e14e3cd3b6d48157ea0706114f7\\">View this gist on GitHub</a></noscript></div>
<!-- /wp:coblocks/gist -->"
`;
